pkg/utils/yamlhelper.go

```diff
@@ -0,0 +1,55 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pmezard/go-difflib/difflib"
+	"gopkg.in/yaml.v2"
+)
+
+// function to get the yaml output of an object
+// marshal/unmarshal to and from json in order to maintain consistency between json and yaml
+// currently k8s objects have json field tags, so this helps in skipping empty fields
+// and give consistent output yaml similar to json
+func getYamlOfObject(object interface{}) (string, error) {
+	if b, err := json.Marshal(object); err != nil {
+		return "", fmt.Errorf("error while encoding %+v into JSON: %s", object, err)
+	} else {
+		var v map[string]interface{}
+		if err := json.Unmarshal(b, &v); err != nil {
+			return "", fmt.Errorf("error while decoding JSON %s into an object: %s", v, err)
+		} else {
+			if y, err := yaml.Marshal(&v); err != nil {
+				return "", fmt.Errorf("error while encoding Map %s into YAML: %s", v, err)
+			} else {
+				return string(y), nil
+			}
+		}
+	}
+}
+
+func generateYamlDiff(yaml1 string, yaml2 string) string {
+	diff, _ := difflib.GetUnifiedDiffString(difflib.UnifiedDiff{
+		A:        difflib.SplitLines(yaml1),
+		B:        difflib.SplitLines(yaml2),
+		FromFile: "Old",
+		FromDate: "",
+		ToFile:   "New",
+		ToDate:   "",
+		Context:  1,
```

Comment:
Just a nit - because YAML object in k8s tend to be rather large and nested, it would be easier to read the diff output if you used a larger number here, e.g. 3. Or perhaps take it as a parameter, up to you.

Reply:
Ack